### PR TITLE
[python] Fix automatic formatting of axes ticks

### PR DESCRIPTION
**Pull request by pluegh**
The changes are tested for all cases using the notebook provided in `examples/inference.ipynb`. The user is warned about potential issues, covering all cases.

### CODE REVIEW

**Comment by pluegh** [comment]
A small update of the example notebooks might also be in order. Nothing broke, but the default usage is slightly easier now. I'll update the PR.

File: python/eos/plot/plotter.py [hunk 1]
```diff
@@ -105,18 +105,21 @@ def setup_plot(self):
...
                 self.xscale = float(myx['scale'])
                 self.xticks = matplotlib.ticker.FuncFormatter(lambda x, pos, xscale=self.xscale: self.xformat.format(x=x / xscale))
-            else:
-                self.xticks = matplotlib.ticker.StrMethodFormatter(self.xformat)
+                self.ax.xaxis.set_major_formatter(self.xticks)
 
-            self.ax.xaxis.set_major_formatter(self.xticks)
+            else:
+                if 'format' in myx:
+                    eos.warn("Argument plot:x:format is only used when plot:x:scale is used")
```
Comment (dvandyk) [inline python/eos/plot/plotter.py:122]: This change should also be reflected in the documentation (``doc/api/python.rst``)

**Review by dvandyk — CHANGES_REQUESTED** [review]
Except for the missing change to the documentation this looks very good!

File: python/eos/plot/plotter.py (hunk 1 above)
Comment (pluegh) [reply python/eos/plot/plotter.py:122]: Done!

**Review by dvandyk — APPROVED** [review]

**Pull request merged**